x-pack/plugins/enterprise_search/public/applications/shared/unsaved_changes_prompt/unsaved_changes_prompt.tsx

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect } from 'react';
+import { Prompt } from 'react-router-dom';
+
+import { i18n } from '@kbn/i18n';
+
+const DEFAULT_MESSAGE_TEXT = i18n.translate('xpack.enterpriseSearch.shared.unsavedChangesMessage', {
+  defaultMessage: 'Your changes have not been saved. Are you sure you want to leave?',
+});
+interface Props {
+  hasUnsavedChanges: boolean;
+  messageText?: string;
+}
+
+export const UnsavedChangesPrompt: React.FC<Props> = ({
+  hasUnsavedChanges,
+  messageText = DEFAULT_MESSAGE_TEXT,
+}) => {
+  useEffect(() => {
+    const handler = (event: BeforeUnloadEvent) => {
+      if (hasUnsavedChanges) {
+        // These 2 lines of code are the recommendation from MDN for triggering a browser prompt for confirming
+        // whether or not a user wants to leave the current site.
```

Comment:
Could be helpful to link the docs here, maybe that recommendation will change in the future!

Reply:
Agreed, that could have been helpful.